app/uk/gov/hmrc/selfservicetimetopay/controllers/ArrangementController.scala

```diff
@@ -44,103 +44,96 @@ class ArrangementController(ddConnector: DirectDebitConnector,
+                        Future.successful(Redirect(routes.ArrangementController.getInstalmentSummary()))
+                      case TTPSubmission(_, _, _, _, _, _, _, _, Some(EligibilityStatus(_, reasons))) =>
+                        Logger.info(s"Failed eligibility check because: $reasons")
+                        Future.successful(Redirect(routes.SelfServiceTimeToPayController.getTtpCallUs()))
                     }
-                  case TTPSubmission(None, _, _, Some(tp@Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(meDebits, _, _, _, _, _), _) =>
-                    sessionCache.put(newSubmission.copy(calculatorData = CalculatorInput(startDate = LocalDate.now(),
-                      endDate = LocalDate.now().plusMonths(3).minusDays(1), debits = tpSA.debits))).map[Result] {
-                      _ => Redirect(routes.CalculatorController.getPaymentToday())
-                    }
-                  case TTPSubmission(_, _, _, Some(tp@Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(meDebits, _, _, _, _, _), _) =>
-                    if (areEqual(tpSA.debits, meDebits)) eligibilityCheck(newSubmission)
-                    else {
-                      sessionCache.put(newSubmission).flatMap[Result] {
-                        _ => Future.successful(Redirect(routes.CalculatorController.getMisalignmentPage()))
-                      }
-                    }
-                  case _ =>
-                    Future.successful(Redirect(routes.SelfServiceTimeToPayController.start()))
-                }
-              })
+                case TTPSubmission(_, _, _, Some(tp@Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(meDebits, _, _, _, _, _), _, _) =>
+                  sessionCache.put(newSubmission).flatMap {
+                    _ => Future.successful(Redirect(routes.CalculatorController.getMisalignmentPage()))
+                  }
+                case _ =>
+                  Future.successful(Redirect(routes.SelfServiceTimeToPayController.start()))
+              }
             }
           })
         }
       }
   }
 
-  def getInstalmentSummary: Action[AnyContent] = AuthorisedSaUser {
-    implicit authContext => implicit request =>
+  def getInstalmentSummary: Action[AnyContent] = AuthorisedSaUser { implicit authContext =>
+    implicit request =>
       authorizedForSsttp {
-        sessionCache.get.flatMap {
-          _.fold(redirectToStart)(ttp => {
-            if (areEqual(ttp.taxpayer.get.selfAssessment.get.debits, ttp.calculatorData.debits)) {
-              Future.successful(Ok(instalment_plan_summary(ttp.schedule.getOrElse(throw new RuntimeException("No schedule data")),
-                createDayOfForm(ttp), signedIn = true)))
-            } else {
-              Future.successful(Redirect(routes.CalculatorController.getMisalignmentPage()))
-            }
-          })
-        }
+        case None => redirectToStart
```

Comment:
As authorizedForSsttp now provides the TTPSubmission data, we can now express usages of it using partial function case statements. rather than mapping/folding